### PR TITLE
Axom + Conduit updates

### PR DESCRIPTION
**Pull request by white238**
* Loosen Axom's variants
* Add shared variant for Axom
* Fix clang/xlf rpath'ing problem on blueos for Axom and Conduit

### CODE REVIEW

File: var/spack/repos/builtin/packages/conduit/package.py [hunk 1]
```diff
@@ -408,18 +408,23 @@ def create_host_config(self, spec, prefix, py_site_pkgs_dir=None):
...
-                    flags += "-Wl,-rpath,{0} -Wl,-rpath,{0}64".format(libdir)
+                    rpaths = "-Wl,-rpath,{0} -Wl,-rpath,{0}64".format(libdir)
+
+                    flags  = "${BLT_EXE_LINKER_FLAGS} -lstdc++ " + rpaths
                     cfg.write(cmake_cache_entry("BLT_EXE_LINKER_FLAGS",
                                                 flags))
+                    if "+shared" in spec:
+                        flags = "${CMAKE_SHARED_LINKER_FLAGS} " + rpaths
+                        cfg.write(cmake_cache_entry(
+                                  "CMAKE_SHARED_LINKER_FLAGS", flags))
```
Comment (cyrush) [inline var/spack/repos/builtin/packages/conduit/package.py:427]: Wow, compilers are being very mean to you -- this looks good, thanks for working this out.

**Review by cyrush — APPROVED** [review]

File: var/spack/repos/builtin/packages/axom/package.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ class Axom(CMakePackage, CudaPackage):
     homepage = "https://github.com/LLNL/axom"
     git      = "https://github.com/LLNL/axom.git"
 
-    version('master', branch='master', submodules=True)
+    version('main', branch='main', submodules=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/axom/package.py:54]: This version will sort as being older than all other stable releases, is this what you want?

Comment (cyrush) [reply var/spack/repos/builtin/packages/axom/package.py:54]: does spack have special support for `master`, could that support also include `main`?

Comment (cyrush) [reply var/spack/repos/builtin/packages/axom/package.py:54]: `master` branch no longer exists ...

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/axom/package.py:54]: We can name the version whatever we want. We currently have support for `develop`, `master`, `head`, and `trunk`. We could add `main`, or we could do:
```suggestion
    version('master', branch='main', submodules=True)
```

Comment (white238) [reply var/spack/repos/builtin/packages/axom/package.py:54]: Spoke to @becker33, added fix from him that added main to the list of blessed branches.

**Review by adamjstewart — APPROVED** [review]

**Comment by white238** [comment]
@adamjstewart  thanks for merging.  This passed all our internal Axom tests except half of them fail when i turn on mfem's mpi variant but I think it might be a compiler+system problem not specific to spack or axom.  I'll try to track down the exact cause when I have more cycles.

**Comment by white238** [comment]
> @adamjstewart thanks for merging. This passed all our internal Axom tests except half of them fail when i turn on mfem's mpi variant but I think it might be a compiler+system problem not specific to spack or axom. I'll try to track down the exact cause when I have more cycles.

Actually loosening the MFEM variants didn't get into this PR so that will come after I have more confidence in it.

**Pull request merged**